src/compiler/moduleSpecifiers.ts

```diff
@@ -282,10 +282,12 @@ namespace ts.moduleSpecifiers {
         const redirects = host.redirectTargetsMap.get(importedPath) || emptyArray;
         const importedFileNames = [...(referenceRedirect ? [referenceRedirect] : emptyArray), importedFileName, ...redirects];
         const targets = importedFileNames.map(f => getNormalizedAbsolutePath(f, cwd));
+        let filterIgnoredPaths = some(targets, not(containsIgnoredPath));

```

Comment:
Alternatively

```suggestion
        let filterIgnoredPaths = !every(targets, containsIgnoredPath);
```

but yours reads better.

I might have called it

`shouldFilterIgnoredPaths` though.